pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/kube_ingress_reconciler.go

```diff
@@ -263,6 +265,56 @@ func semanticIngressEquals(desired, existing *netv1.Ingress) bool {
 	return equality.Semantic.DeepEqual(desired.Spec, existing.Spec)
 }
 
+func getRouteURLIfExists(cli client.Client, isvc *v1beta1.InferenceService) (*apis.URL, error) {
+	foundRoute := false
+	routeReady := false
+	route := &routev1.Route{}
+	err := cli.Get(context.TODO(), types.NamespacedName{Name: isvc.Name, Namespace: isvc.Namespace}, route)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check if the route is owned by the InferenceService
+	for _, ownerRef := range route.OwnerReferences {
+		if ownerRef.UID == isvc.UID {
+			foundRoute = true
+		}
+	}
+
+	// Check if the route is admitted
+	for _, ingress := range route.Status.Ingress {
+		for _, condition := range ingress.Conditions {
+			if condition.Type == "Admitted" && condition.Status == "True" {
```

Comment:
There might be a constant from netv1 for "Admitted"

Reply:
is there a constant as Yuan suggested?

Reply:
updates on this?